src/Driver/AmazonSqsDriver.php

```diff
@@ -101,28 +103,38 @@ public function send(MessageInterface $message): bool
     public function wait(Closure $callback): void
     {
         while (true) {
-            $result = $this->client->receiveMessage(array(
+            if (!$this->shouldProcessNext()) {
+                break;
+            }
+            $this->checkRestart();
```

Comment:
I would put `checkRestart()` as first check. You want to check if restart is needed before checking if you should process next item.

In case `shouldProcessNext()` gets broken (whatever reasones) and it will return always false, it would never reach restart statment.